pallets/subtensor/src/subnets/subnet.rs

```diff
@@ -0,0 +1,352 @@
+        Self::set_network_last_lock(actual_tao_lock_amount);
+
+        // --- 8. Set initial and custom parameters for the network.
+        let default_tempo = DefaultTempo::<T>::get();
+        Self::init_new_network(netuid_to_register, default_tempo);
+        log::debug!("init_new_network: {:?}", netuid_to_register);
+
+        // --- 9 . Add the caller to the neuron set.
+        Self::create_account_if_non_existent(&coldkey, hotkey);
+        Self::append_neuron(netuid_to_register, hotkey, current_block);
+        log::debug!(
+            "Appended neuron for netuid {:?}, hotkey: {:?}",
+            netuid_to_register,
+            hotkey
+        );
+
+        // --- 10. Set the mechanism.
+        SubnetMechanism::<T>::insert(netuid_to_register, mechid);
+        log::debug!(
+            "SubnetMechanism for netuid {:?} set to: {:?}",
+            netuid_to_register,
+            mechid
+        );
+
+        // --- 11. Set the creation terms.
+        NetworkLastRegistered::<T>::set(current_block);
+        NetworkRegisteredAt::<T>::insert(netuid_to_register, current_block);
+
+        // --- 14. Init the pool by putting the lock as the initial alpha.
+        TokenSymbol::<T>::insert(
+            netuid_to_register,
+            Self::get_symbol_for_subnet(netuid_to_register),
+        ); // Set subnet token symbol.
+
+        // Put initial TAO from lock into subnet TAO and produce numerically equal amount of Alpha
+        // The initial TAO is the locked amount, with a minimum of 1 RAO and a cap of 100 TAO.
+        let pool_initial_tao = Self::get_network_min_lock();
+        let actual_tao_lock_amount_less_pool_tao =
+            actual_tao_lock_amount.saturating_sub(pool_initial_tao);
+        SubnetTAO::<T>::insert(netuid_to_register, pool_initial_tao);
+        SubnetAlphaIn::<T>::insert(netuid_to_register, pool_initial_tao);
+        SubnetOwner::<T>::insert(netuid_to_register, coldkey.clone());
+        SubnetOwnerHotkey::<T>::insert(netuid_to_register, hotkey.clone());
+        TotalStakeAtDynamic::<T>::insert(netuid_to_register, TotalStake::<T>::get());
+
+        if actual_tao_lock_amount_less_pool_tao > 0 {
+            Self::burn_tokens(actual_tao_lock_amount_less_pool_tao);
+        }
+
+        // --- 15. Add the identity if it exists
+        if let Some(identity_value) = identity {
+            ensure!(
+                Self::is_valid_subnet_identity(&identity_value),
+                Error::<T>::InvalidIdentity
+            );
+
+            SubnetIdentitiesV2::<T>::insert(netuid_to_register, identity_value);
+            Self::deposit_event(Event::SubnetIdentitySet(netuid_to_register));
+        }
+
+        // --- 16. Emit the NetworkAdded event.
+        log::info!(
+            "NetworkAdded( netuid:{:?}, mechanism:{:?} )",
+            netuid_to_register,
+            mechid
+        );
+        Self::deposit_event(Event::NetworkAdded(netuid_to_register, 0));
```

Comment:
Should this be 0 else all new networks aren't set to be part of the dTao mechanism?

Reply:
thanks. will add into `devnet-ready` for the next runtime upgrade (after dTAO)